### PR TITLE
Fraud detection algos

### LINKED ISSUE #188: [demo] Fraud detection algos
Use Spark LightGBM to train a binary classifier for fraud detection  based on [ULB credit card dataset](https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud).

### PR DESCRIPTION
**Pull request by intelligencegear**
We train a binary classifier using Spark LightGBM on ULB credit card dataset and use `AUC/KS/ConfusionMatrix` to evaluate the performance of this model. Moreover, we provide the demo of SMOTE and Isolation Forest in notebooks.

close #188

### CODE REVIEW

File: demo/dataset/README-CN.md [hunk 1]
```diff
@@ -158,3 +159,13 @@ python fg_large_dataset.py --conf fg_large_dataset.yaml.dev
 ### Feature Generation
 下载完成后，我们可以使用 [fg.ipynb](./tianchi_loan/fg.ipynb) 来生成我们模型中使用的数值特征。
 
+
+## ULB-CreditCard
+在这一节中，我们将介绍如何 ULB 提供的 [信用卡欺诈检测数据集](https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud)的中数据。
```
Comment (wikty) [inline demo/dataset/README-CN.md:164]: 感觉像少了几个字

File: demo/dataset/README-CN.md [hunk 2]
```diff
@@ -158,3 +159,13 @@ python fg_large_dataset.py --conf fg_large_dataset.yaml.dev
...
+
+### Download Data
+首先我们应该从 [ULB 信用卡欺诈数据在 Kaggle 的主页](https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud)上手动下载这个数据集。
+
+### Feature Generation
+下载完成后，我们可以使用 [fg.ipynb](./tianchi_loan/fg.ipynb) 来生成我们模型中使用的数值特征并使用 [SMOTE](https://imbalanced-learn.org/stable/references/generated/imblearn.over_sampling.SMOTE.html) 算法对正样本进行上采样。
```
Comment (wikty) [inline demo/dataset/README-CN.md:170]: 这个fg路径应该不是tianchi吧

Comment (andafterall) [reply demo/dataset/README-CN.md:170]: 嗯嗯，这里应该是写错了。
ULB-CreditCard 用的 [fg.ipynb](./tianchi_loan/fg.ipynb)

Comment (intelligencegear) [reply demo/dataset/README-CN.md:170]: I will fix it.

File: demo/riskmodels/fraud_detection/spark_lgbm.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    return onnx_ml
+
+def onnx_transform(onnx_ml, test_data):
+    result = onnx_ml.transform(test_data)
+    print("ONNX transform sample:", result.limit(10).toPandas())
+
+if __name__=="__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--conf', type=str, action='store', default='', help='config file path')
+    args = parser.parse_args()
```
Comment (andafterall) [inline demo/riskmodels/fraud_detection/spark_lgbm.py:163]: Can we put parser logic into load_config func ?

File: demo/riskmodels/fraud_detection/spark_lgbm.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    booster_model_str = model.getLightGBMBooster()
+    booster_model_str = booster_model_str.modelStr()
+    booster_model_str = booster_model_str.get()
+    booster = lgb.Booster(model_str = booster_model_str)
+    onnx_model = convert_model(booster, len_data_columns)
+    return onnx_model
+
+def save_onnx_model(onnx_model, model_onnx_path, **kwargs):
+    onnxmltools.utils.save_model(onnx_model,'lightgbm.onnx')
+    loaded_model = onnxmltools.utils.load_model('lightgbm.onnx')
```
Comment (andafterall) [inline demo/riskmodels/fraud_detection/spark_lgbm.py:134]: It's better to take load logic out of save_onnx_model func.

File: demo/riskmodels/fraud_detection/spark_lgbm.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    ## write evaluation to s3
+    write_dataset_to_s3(predictions, **params)
+    ## convert model to onnx format & load the model
+    print("Debug -- transform lightgbm model into ONNX format...") 
+    onnx_model = get_onnx_model(model,len(train_dataset.columns)-1)
+    loaded_model = save_onnx_model(onnx_model, **params)
+    onnx_ml = launch_onnx_model(loaded_model)
+    ## predicting using onnx lightgbm model
+    onnx_ml = load_onnx(onnx_ml)
```
Comment (andafterall) [inline demo/riskmodels/fraud_detection/spark_lgbm.py:203]: Merge load_onnx and launch_onnx_model into one single func.

File: demo/riskmodels/fraud_detection/spark_lgbm.py (hunk 4 above)
Comment (intelligencegear) [reply demo/riskmodels/fraud_detection/spark_lgbm.py]: fixed

File: demo/riskmodels/fraud_detection/spark_lgbm.py (hunk 5 above)
Comment (intelligencegear) [reply demo/riskmodels/fraud_detection/spark_lgbm.py]: fixed

File: demo/riskmodels/fraud_detection/spark_lgbm.py (hunk 3 above)
Comment (intelligencegear) [reply demo/riskmodels/fraud_detection/spark_lgbm.py:159]: I think it may be better parse python input parameters outside.

**Review by dmetasoul01 — APPROVED** [review]

**Pull request merged**